modules/configuration/ajax/process.php

```diff
@@ -18,22 +22,38 @@
 require_once 'NDB_Client.class.inc';
 require_once "Utility.class.inc";
 
-$user =& User::singleton();
-if (!$user->hasPermission('config')) {
-    header("HTTP/1.1 403 Forbidden");
-    exit;
+if (!\User::singleton()->hasPermission('config')) {
+    http_response_code(403);
+    return;
 }
 
 $client = new NDB_Client();
-$client->makeCommandLine();
 $client->initialize();
 
-$DB =& Database::singleton();
+$DB      = \Database::singleton();
+$pathIDs = getPathIDs();
 foreach ($_POST as $key => $value) {
     if (is_numeric($key)) { //update
```

Comment:
When going from a blank field and entering an invalid path, update is successful. (see screenshot)

I think the issue is that validation of path only happens if the key is numeric. Though `configuration_helper.js` has a function which concatenates the prefix `add-` to the ID key (making it non-numeric)

Try printing the POST request to your error log to see the case of the add- prefix
EX:
`Array\n(\n    [16] => /var/www/loris/\n    [21] => tools/logs/\n    [add-22] => \n    `
.... `[add-65] => helloWorld/\n` ....


![Screenshot from 2019-05-06 15-12-13](https://user-images.githubusercontent.com/34260251/57249706-a5cee500-7013-11e9-9df9-c2f2f155a39d.png)